pytorch_lightning/trainer/connectors/profiler_connector.py

```diff
@@ -11,16 +11,30 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License
-from pytorch_lightning.profiler import PassThroughProfiler, SimpleProfiler
+
+from typing import Union
+
+from pytorch_lightning.profiler import BaseProfiler, PassThroughProfiler, SimpleProfiler, AdvancedProfiler
+from pytorch_lightning.utilities import rank_zero_warn
 
 
 class ProfilerConnector:
 
     def __init__(self, trainer):
         self.trainer = trainer
 
-    def on_trainer_init(self, profiler):
+    def on_trainer_init(self, profiler: Union[BaseProfiler, bool, str]):
+
+        if isinstance(profiler, bool):
+            rank_zero_warn("Passing a bool value as a `profiler` argument to Trainer is deprecated"
+                           " and will be removed in v0.11.0. Use str ('simple' or 'advanced') instead.",
+                           DeprecationWarning)
         # configure profiler
-        if profiler is True:
+        if profiler is True or profiler == "simple":
             profiler = SimpleProfiler()
+        elif profiler == "advanced":
+            profiler = AdvancedProfiler()
+        elif isinstance(profiler, str):
+            raise ValueError('when passing string value for the `profiler` parameter of'
+                             ' `Trainer`, it can only be `simple` or `advanced`')
```

Comment:
```suggestion
            raise ValueError("When passing string value for the `profiler` parameter of"
                             " `Trainer`, it can only be 'simple' or 'advanced'")
```